packages/veto-app/frontend/src/hooks/useSettings.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react'
+import gun from '../api/gun'
+
+export type Settings = {
+  darkmode?: boolean
+  passwordConfirm?: boolean
+}
+
+export default (): [Settings, Function] => {
+  const db = gun.get('settings')
+  const [settings, setSettings] = useState<Settings>({})
+
+  useEffect(() => {
+    db.map().on((value, key) => {
+      setSettings((settings) => ({
+        ...settings,
+        [key]: value,
+      }))
+    })
+
+    return () => {
+      db.map().off()
```

Comment:
Is this a gun specific thing? Can we get a link to relevant documentation?

Reply:
added a link to the docs